alicloud/service_alicloud_polardb.go

```diff
@@ -0,0 +1,811 @@
+		raw, err := s.client.WithPolarDBClient(func(polarDBClient *polardb.Client) (interface{}, error) {
+			return polarDBClient.GrantAccountPrivilege(request)
+		})
+		if err != nil {
+			if IsExceptedErrors(err, OperationDeniedDBStatus) {
+				return resource.RetryableError(err)
+			}
+			return resource.NonRetryableError(err)
+		}
+
+		addDebug(request.GetActionName(), raw, request.RpcRequest, request)
+
+		return nil
+	})
+
+	if err != nil {
+		return WrapErrorf(err, DefaultErrorMsg, id, request.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+
+	if err := s.WaitForPolarDBAccountPrivilege(id, dbName, Available, DefaultTimeoutMedium); err != nil {
+		return WrapError(err)
+	}
+
+	return nil
+}
+
+func (s *PolarDBService) RevokePolarDBAccountPrivilege(id, dbName string) error {
+	parts, err := ParseResourceId(id, 3)
+	if err != nil {
+		return WrapError(err)
+	}
+	request := polardb.CreateRevokeAccountPrivilegeRequest()
+	request.RegionId = s.client.RegionId
+	request.DBClusterId = parts[0]
+	request.AccountName = parts[1]
+	request.DBName = dbName
+
+	err = resource.Retry(3*time.Minute, func() *resource.RetryError {
+		raw, err := s.client.WithPolarDBClient(func(polarDBClient *polardb.Client) (interface{}, error) {
+			return polarDBClient.RevokeAccountPrivilege(request)
+		})
+		if err != nil {
+			if IsExceptedErrors(err, OperationDeniedDBStatus) {
+				return resource.RetryableError(err)
+			}
+			return resource.NonRetryableError(err)
+		}
+
+		addDebug(request.GetActionName(), raw, request.RpcRequest, request)
+
+		return nil
+	})
+
+	if err != nil {
+		return WrapErrorf(err, DefaultErrorMsg, id, request.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+
+	if err := s.WaitForPolarDBAccountPrivilegeRevoked(id, dbName, DefaultTimeoutMedium); err != nil {
+		return WrapError(err)
+	}
+
+	return nil
+}
+
+func (s *PolarDBService) WaitForPolarDBAccountPrivilegeRevoked(id, dbName string, timeout int) error {
+	deadline := time.Now().Add(time.Duration(timeout) * time.Second)
+	for {
+		object, err := s.DescribePolarDBAccountPrivilege(id)
+		if err != nil {
+			return WrapError(err)
+		}
+
+		exist := false
+		if object != nil {
+			for _, dp := range object.DatabasePrivileges {
+				if dp.DBName == dbName {
+					exist = true
+					break
+				}
+			}
+		}
+
+		if !exist {
+			break
+		}
+		if time.Now().After(deadline) {
+			return WrapErrorf(err, WaitTimeoutMsg, id, GetFunc(1), timeout, "", dbName, ProviderERROR)
+		}
+
+	}
+	return nil
+}
+
+func (s *PolarDBService) WaitForPolarDBAccountPrivilege(id, dbName string, status Status, timeout int) error {
+	parts, err := ParseResourceId(id, 3)
+	if err != nil {
+		return WrapError(err)
+	}
+	deadline := time.Now().Add(time.Duration(timeout) * time.Second)
+	for {
+		object, err := s.DescribePolarDBAccountPrivilege(id)
+		if err != nil {
+			if NotFoundError(err) || IsExceptedErrors(err, []string{InvalidDBClusterIdNotFound, InvalidDBClusterNameNotFound}) {
```

Comment:
The line 255 error comes from a service method and this error can be `NotFoundError`. So, the line 227 should be `if NotFoundError(err) {`